test/spec_utils.rb

```diff
@@ -257,30 +257,38 @@ def initialize(*)
     Rack::Utils.build_nested_query('foo' => 'bar', 'baz' => {}).
       should.equal 'foo=bar'
 
-    # The ordering of the output query string is unpredictable with 1.8's
-    # unordered hash. Test that build_nested_query performs the inverse
-    # function of parse_nested_query.
-    [{"foo" => nil, "bar" => ""},
-      {"foo" => "bar", "baz" => ""},
-      {"foo" => ["1", "2"]},
-      {"foo" => "bar", "baz" => ["1", "2", "3"]},
-      {"foo" => ["bar"], "baz" => ["1", "2", "3"]},
-      {"foo" => ["1", "2"]},
-      {"foo" => "bar", "baz" => ["1", "2", "3"]},
-      {"x" => {"y" => {"z" => "1"}}},
-      {"x" => {"y" => {"z" => ["1"]}}},
-      {"x" => {"y" => {"z" => ["1", "2"]}}},
-      {"x" => {"y" => [{"z" => "1"}]}},
-      {"x" => {"y" => [{"z" => ["1"]}]}},
-      {"x" => {"y" => [{"z" => "1", "w" => "2"}]}},
-      {"x" => {"y" => [{"v" => {"w" => "1"}}]}},
-      {"x" => {"y" => [{"z" => "1", "v" => {"w" => "2"}}]}},
-      {"x" => {"y" => [{"z" => "1"}, {"z" => "2"}]}},
-      {"x" => {"y" => [{"z" => "1", "w" => "a"}, {"z" => "2", "w" => "3"}]}}
-    ].each { |params|
-      qs = Rack::Utils.build_nested_query(params)
-      Rack::Utils.parse_nested_query(qs).should.equal params
-    }
```

Comment:
@tenderlove I agree with what you are saying, but I think that this way the test acts as documentation. -and it also test against the actual result which can detect bugs that might be in ``` parse_nested_query ```-.
Also, the opposite scenario of each of those tests is tested for ``` parse_nested_query ```. Maybe we can add one test case that documents how ``` build_nested_query ``` and ``` parse_nested_query ```work together if we want to be explicit. what do you think?

Anyway, I can remove that commit from the PR if you think that is not worth it and just keep the other one :smiley: